nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/versioning/VersioningService.java

```diff
@@ -106,14 +107,16 @@ boolean isPreSaveDoingCheckOut(Document doc, boolean isDirty, VersioningOption o
     /**
      * Applies versioning options before document save.
      *
+     * @param session the core session
      * @param doc the document
      * @param isDirty {@code true} if there is actual data to save
      * @param option an option chosen by the user or framework
      * @param checkinComment a checkin comment
      * @param context map event info
      * @return the validated option (to use in doPostSave)
+     * @since 9.3
      */
-    VersioningOption doPreSave(Document doc, boolean isDirty, VersioningOption option, String checkinComment,
+    VersioningOption doPreSave(CoreSession session, Document doc, boolean isDirty, VersioningOption option, String checkinComment,
```

Comment:
You can't change the method signature, this breaks backward compatibility, unless it's very low level API.
Furthermore, as HFs only replace some bundles, this could lead to issue.